saviSimulation/savi/jason_processing/UxVBehavior.java

```diff
@@ -180,14 +180,14 @@ public void update(UxV uxv, double simTime, int perceptionDistance, List<WorldOb
 	 */
 	protected void updatePercepts(PVector mypos) {
 		PerceptionSnapshot P = new PerceptionSnapshot();
-		PVector myposWithError = new PVector();
-		
+
 		//if position sensor is failing will return an error
-		myposWithError = getPositionWithError(mypos,sensorsErrorProb);
+		PVector myposWithError = getPositionWithError(mypos,sensorsErrorProb);
 		
 		P.addPerception(new PositionPerception(this.time, (double)myposWithError.x, (double)myposWithError.y, (double)myposWithError.z));
+
 		//Add velocity
-		P.addPerception(new VelocityPerception(this.time, Math.atan(myposWithError.x/myposWithError.y), 0, this.speedVal));			
+		P.addPerception(new VelocityPerception(this.time, this.compasAngle, 0, this.speedVal));
```

Comment:
I think the other line was correct. Otherwise, you do not introduce error in the compass angle
@gtrabes  can you please check?

Reply:
I think the position perceived should be independent of the compasAngle, that's why i propose a way to introduce error on the compassAngle on my early comment.

Reply:
Sorry, I didn't see that comment. I went directly to review the code